bundles/org.openhab.transform.basicprofiles/src/main/java/org/openhab/transform/basicprofiles/internal/profiles/InactivityProfile.java

```diff
@@ -133,54 +108,72 @@ public ProfileTypeUID getProfileTypeUID() {
 
     @Override
     public void onStateUpdateFromItem(State itemState) {
-        logger.trace("onStateUpdateFromItem() item:{}, state:{}", itemChannelLink.getItemName(), itemState);
+        if (!itemState.equals(targetState)) {
+            logger.trace("onStateUpdateFromItem: {}", itemChannelLink);
+            cancelTimeoutTask();
+            rescheduleTimeoutTask();
+        }
     }
 
     @Override
     public void onCommandFromItem(Command itemCommand) {
-        logger.trace("onCommandFromItem() item:{}, command:{}", itemChannelLink.getItemName(), itemCommand);
+        if (!itemCommand.equals(targetState)) {
+            logger.trace("onCommandFromItem: {}", itemChannelLink);
+            cancelTimeoutTask();
+            rescheduleTimeoutTask();
+        }
     }
 
     @Override
     public void onCommandFromHandler(Command handlerCommand) {
         cancelTimeoutTask();
-        if (itemChannelLinked()) {
-            Command itemCommand = OnOffType.from(inverted);
-            logger.debug("onCommandFromHandler() item:{}, itemCommand:{}, handlerCommand:{}",
-                    itemChannelLink.getItemName(), handlerCommand, itemCommand);
-            callback.sendCommand(itemCommand);
+        if (!removed) {
```

Comment:
It's not necessary to do this check here.
Once the link is removed, openhab will not call any of this link instance's handlers.

And because the schedule is cancelled in the registry listener's `removed` and `updated`, the `removed` is also not needed in the onTimeout(), so it can be removed completely.

Reply:
I know you said that before. So I had removed it. However when I was doing the live tests with commenting out and restoring the Item in the .items file, I did encounter a case where the binding did try to update the Item during the interval between the old items being destroyed and the new ones being loaded. It caused a log WARN that the binding was trying to update a non existing Item. It is not a big deal, and probably only occurs once in a blue moon, however I added back this check to avoid shocking users.